lib/Raven/Middleware/ModulesMiddleware.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Raven\Middleware;
+
+use Composer\Factory;
+use Composer\IO\NullIO;
+use Psr\Http\Message\ServerRequestInterface;
+use Raven\Configuration;
+use Raven\Event;
+
+/**
+ * This middleware logs with the event details all the versions of the packages
+ * installed with Composer, if any.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class ModulesMiddleware
+{
+    /**
+     * @var Configuration The Raven client configuration
+     */
+    private $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The Raven client configuration
+     */
+    public function __construct(Configuration $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * Collects the needed data and sets it in the given event object.
+     *
+     * @param Event                       $event     The event being processed
+     * @param callable                    $next      The next middleware to call
+     * @param ServerRequestInterface|null $request   The request, if available
+     * @param \Exception|\Throwable|null  $exception The thrown exception, if available
+     * @param array                       $payload   Additional data
+     *
+     * @return Event
+     */
+    public function __invoke(Event $event, callable $next, ServerRequestInterface $request = null, $exception = null, array $payload = [])
+    {
+        $composerFilePath = $this->config->getProjectRoot() . DIRECTORY_SEPARATOR . 'composer.json';
+
+        if (file_exists($composerFilePath)) {
+            $composer = Factory::create(new NullIO(), $composerFilePath, true);
+            $locker = $composer->getLocker();
+
+            if ($locker->isLocked()) {
+                $modules = [];
+
+                foreach ($locker->getLockedRepository()->getPackages() as $package) {
+                    $modules[$package->getName()] = $package->getVersion();
```

Comment:
It's better to use the `getPrettyVersion()` here, it has a more readable format

Reply:
`getPrettyVersion()` formats the version in a different way, e.g. for the `clue/stream-filter` package the version `1.4.0` would appear as `v1.4.0` (this is what is tagged on GitHub) vs `1.4.0.0`. Basically `getVersion` normalizes the string and ensures that the format is consistent for all packages regardeless of how the version was really named. Wouldn't it be better to use this standard format?